src/core/halide2isl.cc

```diff
@@ -82,44 +82,141 @@ isl::aff makeIslAffFromInt(isl::space space, int64_t val) {
   return isl::aff(isl::local_space(space), v);
 }
 
-isl::aff makeIslAffFromExpr(isl::space space, const Expr& e) {
+std::vector<isl::aff> makeIslAffBoundsFromExpr(
+    isl::space space,
+    const Expr& e,
+    bool allowMin,
+    bool allowMax);
+
+namespace {
+/*
+ * Convert Halide binary expression "op" into a list of isl affine functions by
+ * converting its LHS and RHS into lists of affs and concatenating those lists.
+ * This is intended to be used with Min/Max operations in upper/lower bound
+ * computations, respectively.  Essentially, this allows for replacements
+ *   x < min(a,min(b,c)) <=> x < a AND x < b AND x < c
+ *   x > max(a,max(b,c)) <=> x > a AND x > b AND x > c
+ */
+template <typename T>
+inline std::vector<isl::aff>
+concatAffs(isl::space space, T op, bool allowMin, bool allowMax) {
+  std::vector<isl::aff> result;
```

Comment:
Optionally you could write it this way with a proper ```operator+=```, but not worth changing, just mentioning.
```
std::vector<isl::aff> result(makeIslAffBoundsFromExpr(space, op->a, allowMin, allowMax));
results += makeIslAffBoundsFromExpr(space, op->b, allowMin, allowMax);
return result;
```

Reply:
Yes, considered that but decided that `+=` in the context where we operate on affine functions with another overloaded `operator+` can be misleading.